### PR TITLE
Remove some Skips

### PR DESCRIPTION
**Pull request by xpivarc**
### What this PR does
We now have less skips

Fixes #

### Release note

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-controller/services/template_test.go [hunk 1]
```diff
@@ -212,6 +212,19 @@ var _ = Describe("Template", func() {
 
 			return vmi
 		}
+		It("should set cmd line argument for software emulation", func() {
```
Comment (ormergi) [inline pkg/virt-controller/services/template_test.go:215]: This seem not related to this commit, right?

File: tests/vmi_lifecycle_test.go [hunk 2]
```diff
@@ -1428,7 +1428,7 @@ var _ = Describe("[rfe_id:273][crit:high][arm64][vendor:cnv-qe@redhat.com][level
 				nodeList := libnode.GetAllSchedulableNodes(kubevirt.Client())
 
 				if len(nodeList.Items) == 0 {
```
Comment (ormergi) [inline tests/vmi_lifecycle_test.go:1430]: Same here, it seems the check can be removed because now CI can filter these tests on unsupported envs

File: tests/vmi_lifecycle_test.go [hunk 3]
```diff
@@ -886,7 +886,7 @@ var _ = Describe("[rfe_id:273][crit:high][arm64][vendor:cnv-qe@redhat.com][level
 				Expect(nodes.Items).ToNot(BeEmpty(), "There should be some compute node")
 				supportedCpuModels = libnode.GetSupportedCPUModels(*nodes)
 				if len(supportedCpuModels) < 2 {
-					Skip("need at least 2 supported cpuModels for this test")
+					Fail("need at least 2 supported cpu models for this test")
```
[the remark this answers is not in the document]
Comment (xpivarc) [reply tests/vmi_lifecycle_test.go:887]: I personally like failing fast. As in some cases it can dramatically help identifying why the test failed (you would usually see timeout error. and would need to deep dive into cluster state)

[the remark this answers is not in the document]
Comment (ormergi) [reply tests/vmi_lifecycle_test.go:887]: If this test is filtered out by CI initially there is no need to fail fast or any troubleshooting :)

[the remark this answers is not in the document]
Comment (xpivarc) [reply tests/vmi_lifecycle_test.go:887]: >If this test is filtered out by CI initially there is no need to fail fast or any troubleshooting :)

That is true but if it is not filtered out it can be difficult to know why they fail...

**Comment by xpivarc** [comment]
> Very nice work!
> 
> Does it necessary to change sig-compute the prow-jobs or the automation/test.sh script following these changes?

Now included, I wanted to see what fails. The fails seemed reasonable

File: automation/test.sh [hunk 4]
```diff
@@ -429,18 +429,18 @@ if [[ -z ${KUBEVIRT_E2E_FOCUS} && -z ${KUBEVIRT_E2E_SKIP} && -z ${label_filter}
   elif [[ $TARGET =~ vgpu.* ]]; then
     label_filter='(VGPU)'
   elif [[ $TARGET =~ sig-compute-realtime ]]; then
-    label_filter='(sig-compute-realtime)'
+    label_filter='(sig-compute-realtime) && !(software-emulation, SEV, SEVES)'
```
Comment (xpivarc) [inline automation/test.sh:432]: @dhiller @brianmcarey Can you have a look at least on the changes in this file?

File: tests/instancetype/instancetype.go [hunk 5]
```diff
@@ -1020,10 +1019,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 		)
 	})
 
-	Context("instance type with dedicatedCPUPlacement enabled", func() {
-		BeforeEach(func() {
-			checks.SkipTestIfNoCPUManager()
-		})
+	Context("instance type with dedicatedCPUPlacement enabled", decorators.RequiresTwoWorkerNodesWithCPUManager, func() {
```
Comment (jean-edouard) [inline tests/instancetype/instancetype.go:1022]: I believe this makes the test more expensive.
It used to require 1 node with CPUManager (i.e. a 2-node cluster), this now makes it require 2 (i.e. a 3-node cluster).

File: tests/network/sriov.go [hunk 6]
```diff
@@ -106,8 +105,7 @@ var _ = Describe("[Serial]SRIOV", Serial, decorators.SRIOV, func() {
 				To(Succeed(), shouldCreateNetwork)
 		})
 
-		It("should have cloud-init meta_data with aligned cpus to sriov interface numa node for VMIs with dedicatedCPUs", func() {
-			checks.SkipTestIfNoCPUManager()
+		It("should have cloud-init meta_data with aligned cpus to sriov interface numa node for VMIs with dedicatedCPUs", decorators.RequiresTwoWorkerNodesWithCPUManager, func() {
```
Comment (jean-edouard) [inline tests/network/sriov.go:108]: Same as above

**Review by jean-edouard — COMMENTED** [review]
Great cleanup, thank you!
I have one concern, which is repeated multiple times (commented in-line on 2 instances). We shouldn't use the 3-node lane (sig-compute-migrations) when only 2 nodes are necessary.

**Review by dhiller — APPROVED** [review]
Thank you @xpivarc - great job in replacing lots of skips by converting them into Ginkgo filters.

Looks good to me!

One question inline.

File: tests/instancetype/instancetype.go (hunk 5 above)
Comment (ormergi) [reply tests/instancetype/instancetype.go:1022]: On CI we spin up 3 nodes cluster aren't we?

Comment (jean-edouard) [reply tests/instancetype/instancetype.go:1022]: I believe only sig-compute-migrations has 3 nodes, the others lanes have 2.

File: tests/vmi_lifecycle_test.go (hunk 2 above)
Comment (xpivarc) [reply tests/vmi_lifecycle_test.go:1430]: Same as the other comment, I think this is still helpful

File: tests/instancetype/instancetype.go (hunk 5 above)
Comment (xpivarc) [reply tests/instancetype/instancetype.go:1022]: Introduced `RequiresNodeWithCPUManager`

File: tests/network/sriov.go (hunk 6 above)
Comment (xpivarc) [reply tests/network/sriov.go:108]: Done

File: automation/test.sh (hunk 4 above)
Comment (dhiller) [reply automation/test.sh:434]: Dumb q: I saw introduction of VSOCK label, however I don't see it used anywhere in this file - is that by intent? Do we want to use that label dynamically somewhere else?

Comment (dhiller) [reply automation/test.sh:434]: ^^ @xpivarc

File: tests/instancetype/instancetype.go [hunk 7]
```diff
@@ -1020,10 +1019,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 		)
 	})
 
-	Context("instance type with dedicatedCPUPlacement enabled", func() {
-		BeforeEach(func() {
-			checks.SkipTestIfNoCPUManager()
-		})
+	Context("instance type with dedicatedCPUPlacement enabled", decorators.RequiresNodeWithCPUManager, func() {
```
Comment (jean-edouard) [inline tests/instancetype/instancetype.go:1023]: Looks like this line upsets the linter

**Review by jean-edouard — COMMENTED** [review]
/approve
@ormergi have your concerns been resolved?

File: tests/instancetype/instancetype.go (hunk 7 above)
Comment (xpivarc) [reply tests/instancetype/instancetype.go]: Weird, should be fixed now

File: automation/test.sh (hunk 4 above)
Comment (xpivarc) [reply automation/test.sh:434]: @dhiller oh and yes I have a place where this will be used but as user of the test suite.

Comment (dhiller) [reply automation/test.sh:434]: Just one note on the SEV: as we currently don't have a functioning cluster there's no way to check whether this will work completely.

**Comment by xpivarc** [comment]
@dhiller PTAL, addressed linter issue
